presto-orc/src/test/java/com/facebook/presto/orc/TestOrcReaderMemoryUsage.java

```diff
@@ -0,0 +1,295 @@
+                }
+
+                assertEquals(reader.nextBatch(), -1);
+                assertEquals(reader.getReaderPosition(), rows);
+                assertEquals(reader.getFilePosition(), reader.getReaderPosition());
+            }
+        }
+    }
+
+    @Test
+    public void testBigIntTypeWithNulls()
+            throws Exception
+    {
+        try (TempFile tempFile = new TempFile()) {
+            int rows = 10000;
+            createSingleColumnFileWithNullValues(tempFile.getFile(), rows);
+
+            try (OrcRecordReader reader = createCustomOrcRecordReader(tempFile, ORC, OrcPredicate.TRUE, BIGINT)) {
+                assertEquals(reader.getReaderRowCount(), rows);
+                assertEquals(reader.getReaderPosition(), 0);
+                assertEquals(reader.getStreamReaderRetainedSizeInBytes(), 0);
+                assertEquals(reader.getSystemMemoryUsage(), 0);
+
+                while (true) {
+                    int batchSize = reader.nextBatch();
+                    if (batchSize == -1) {
+                        break;
+                    }
+
+                    System.out.println("batchSize: " + batchSize);
+                    Block block = reader.readBlock(BIGINT, 0);
+                    System.out.println("block.getPositionCount(): " + block.getPositionCount());
+
+                    assertEquals(block.getPositionCount(), batchSize);
+
+                    // batchSize could be growing, and StreamReader local buffer size is also growing.
+                    // Here we only verify the memory usage when the batchSize has grown to MAX_BATCH_SIZE.
+                    if (batchSize < MAX_BATCH_SIZE) {
+                        continue;
+                    }
+                    System.out.println("reader.getStreamReaderRetainedSizeInBytes() " + reader.getStreamReaderRetainedSizeInBytes());
+                    System.out.println("reader.getCurrentStripeRetainedSizeInBytes() " + reader.getCurrentStripeRetainedSizeInBytes());
+
+                    assertEquals(reader.getStreamReaderRetainedSizeInBytes(), batchSize * SIZE_OF_BYTE);
+                    assertBetweenInclusive(reader.getCurrentStripeRetainedSizeInBytes(), 3600L, 3700L);
+                    assertEquals(reader.getStreamReaderRetainedSizeInBytes() + reader.getCurrentStripeRetainedSizeInBytes(), reader.getSystemMemoryUsage());
+                }
+
+                assertEquals(reader.nextBatch(), -1);
+                assertEquals(reader.getReaderPosition(), rows);
+                assertEquals(reader.getFilePosition(), reader.getReaderPosition());
+            }
+        }
+    }
+
+    @Test
+    public void testMapTypeWithNulls()
+            throws Exception
+    {
+        Type keyType = BIGINT;
+        Type valueType = BIGINT;
+
+        TypeManager typeManager = new TypeRegistry();
+        FunctionRegistry registry = new FunctionRegistry(typeManager, new BlockEncodingManager(typeManager), new FeaturesConfig());
```

Comment:
Inside of the constructor of FunctionRegistry, it registers typeManager with this newed FunctionRegistry but yes the registry was not used. Anyways I moved this TypeManager to a static member in case it will be used future.